coinlib_flutter/windows/CMakeLists.txt

```diff
@@ -0,0 +1,24 @@
+# The Flutter tooling requires that developers have a version of Visual Studio
+# installed that includes CMake 3.14 or later. You should not increase this
+# version, as doing so will cause the plugin to fail to compile for some
+# customers of the plugin.
+cmake_minimum_required(VERSION 3.14)
+
+# Project-level configuration.
+set(PROJECT_NAME "coinlib_flutter")
+project(${PROJECT_NAME} LANGUAGES CXX)
+
+# Invoke the build for native code shared with the other target platforms.
+# This can be changed to accommodate different builds.
+add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/../src" "${CMAKE_CURRENT_BINARY_DIR}/shared")
+
+# List of absolute paths to libraries that should be bundled with the plugin.
+# This list could contain prebuilt libraries, or libraries created by an
+# external build triggered from this build file.
+set(coinlib_flutter_bundled_libraries
+  # Defined in ../src/CMakeLists.txt.
+  # This can be changed to accommodate different builds.
+  $<TARGET_FILE:coinlib_flutter>
+  ${CMAKE_BINARY_DIR}/../src/secp256k1/lib/src/libsecp256k1-2.dll
```

Comment:
It looks like this is being taken from the `build_windows.dart` script output when it should be possible to build through the flutter build process using CMake allowing the plugin to be used directly for applications without any required additional build script.

The `coinlib_flutter/src/CMakeLists.txt` file provides a CMake build for Linux (inc. Android) . It is included above with `add_subdirectory` but it isn't used by Windows. Replacing that with the `coinlib_flutter/src/secp256k1/` directory and the `coinlib_flutter/src/secp256k1/cmake/x86_64-w64-mingw32.toolchain.cmake` toolchain might work instead.

However the ideal would be to remove the embedded `coinlib_flutter/src/secp256k1` directory and use `ExternalProject_Add` like in `coinlib_flutter/src/CMakeLists.txt`. The only reason why secp256k1 was previously included is for the macOS and iOS builds which do not support CMake. If they can support CMake in the future, that would be preferable and secp256k1 could be removed from the repository in favour of `ExternalProject_Add`.

Maybe it is possible to use `ExternalProject_Add` to build for windows. Either the `coinlib_flutter/src/CMakeLists.txt`  file can be adapted to support Windows or a CMake configuration can be written perhaps directly to `coinlib_flutter/windows/CMakeLists.txt` (this file). Using the `coinlib_flutter/src/secp256k1/CMakeLists.txt` via `add_subdirectory` would be acceptable for now.

Reply:
Partially resolved by 06d2899c8f04f4b8a301f0e71a026cf9ade22ec0, by moving the DLL to and using it from `build/secp256k1.dll` instead of its build script output.  Still plenty left to do to resolve this change request re: using CMake.  Using CMake on Windows has been giving me issues lately, but I'll try to use a pure-CMake approach after resolving the other issues: it's always nice when things to build automagically for the most part.